x/crosschain/keeper/keeper_pending_nonces.go

```diff
@@ -74,13 +74,31 @@ func (k Keeper) RemovePendingNonces(ctx sdk.Context, pendingNonces types.Pending
 
 // utility
 func (k Keeper) RemoveFromPendingNonces(ctx sdk.Context, tssPubkey string, chainID int64, nonce int64) {
-	p, found := k.GetPendingNonces(ctx, tssPubkey, uint64(chainID))
+	p, found := k.GetPendingNonces(ctx, tssPubkey, chainID)
 	if found && nonce >= p.NonceLow && nonce <= p.NonceHigh {
 		p.NonceLow = nonce + 1
 		k.SetPendingNonces(ctx, p)
 	}
 }
 
+func (k Keeper) PendingNoncesByChain(c context.Context, req *types.QueryPendingNoncesByChainRequest) (*types.QueryPendingNoncesByChainResponse, error) {
```

Comment:
This is already implemented here : https://github.com/zeta-chain/node/pull/1131

Reply:
Nice. Do you want to merge your PR into mine or you want to merge it into develop first?